Functions/QualityControl/xASL_qc_CollectParameters.m

```diff
@@ -34,37 +35,43 @@
     error('Missing iSubject...');
 end
 
+% It is recommended to provide session number for ASL
+if nargin<4 || isempty(iSession)
+	iSession = 1;
+	if strcmp(ScanType, 'ASL')
+		warning('Missing iSession for ASL, setting to 1');
+	end
+end
+
 if ~isfield(x,'Output')
     x.Output  = struct; 
 end
-if ~isfield(x.Output,ScanType)
+if ~isfield(x.Output, ScanType)
     x.Output.(ScanType) = struct; 
 end
 
+%% -----------------------------------------------------------------------------------------------
+%% Collect subject-specific (i.e. structural/anatomical) QC results
+
 switch ScanType
     case 'Structural'
-        QCCollectFunction = @xASL_qc_CollectQC_Structural;
+        x = xASL_qc_CollectQC_Structural(x, iSubject);
```

Comment:
Nice simplification of my code :) I can't remember why I used the @function saving, perhaps I thought of using this QC function at multiple locations in the code but I cannot remember seeing it anywhere else indeed :)

Reply:
Actually - you had it written nicely. In switch, you assigned the function handle and then only later you have used it. But now that we require sessions for ASL, I preferred to write it without function handles to avoid any issues with the optional parameters. But this might further change with the required changes for `DWI` and `func` :D

Reply:
I see